eo-maven-plugin/src/test/java/org/eolang/maven/optimization/OptCachedTest.java

```diff
@@ -50,73 +49,92 @@ final class OptCachedTest {
     /**
      * Test case for XML program in cache.
      *
-     * @param tmp Temp dir
+     * @param cache Temp cache dir.
+     * @param dir Temp program dir
      * @throws IOException if I/O fails
      * @todo #2422:60min returnsFromCacheIfXmlAlreadyInCache: this test is unstable.
      *  We should resolve issues with unstable failures and only
      *  then enable the test.
      *  Also, see this <a href="https://github.com/objectionary/eo/issues/2727">issue</a>.
+     * @todo #2790:30min There is repeating logic in tests.
+     *  Code logic repeats in {@link OptCacedTest},
+     *  {@link OptimizeMojoTest.getAlreadyOptimizedResultsFromCache}
+     *  and {@link ShakeMojoTest.getAlreadyShakenResultsFromCache}.
+     *  To check whether a program is loaded from the cache,
+     *  different tests use the same code
+     *  { @code Files.setLastModifiedTime(
+     *  res,
+     *  FileTime.fromMillis(System.currentTimeMillis() + time)
+     *  ); }.
+     *  We need to think about how to remove this duplication.
      */
-    @Disabled
     @Test
-    void returnsFromCacheIfXmlAlreadyInCache(@TempDir final Path tmp) throws IOException {
-        final XML program = OptCachedTest.program(ZonedDateTime.now());
-        OptCachedTest.save(tmp, program);
+    void returnsFromCacheIfXmlAlreadyInCache(@TempDir final Path cache, @TempDir final Path dir)
+        throws Exception {
+        final XML xml = OptCachedTest.program();
+        final Path program = OptCachedTest.save(dir, xml);
+        OptCachedTest.save(cache, xml);
         MatcherAssert.assertThat(
             "We expected that the program will be returned from the cache.",
             new OptCached(
                 path -> {
                     throw new IllegalStateException("This code shouldn't be executed");
                 },
-                tmp
-            ).apply(program),
-            Matchers.equalTo(program)
+                cache,
+                program
+            ).apply(xml),
+            Matchers.equalTo(xml)
         );
     }
 
-    @Disabled
     @Test
-    void returnsFromCacheButTimesSaveAndExecuteDifferent(@TempDir final Path tmp)
-        throws IOException {
-        final XML program = OptCachedTest.program(ZonedDateTime.now().minusMinutes(2));
-        OptCachedTest.save(tmp, program);
+    void returnsFromCacheButLastModifiedTimesDifferent(
+        @TempDir final Path cache,
+        @TempDir final Path dir
+    )
+        throws Exception {
+        final XML xml = OptCachedTest.program();
+        final Path program = OptCachedTest.save(dir, xml);
```

Comment:
@Yanich96 Can you inline this variable, please?
https://www.yegor256.com/2015/09/01/redundant-variables-are-evil.html